contracts/src/SimplePDPService.sol

```diff
@@ -142,37 +134,51 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
     // it also checks that proofs are not late and emits a fault record if so
     function posessionProven(uint256 proofSetId, uint256 challengedLeafCount, uint256 seed, uint256 challengeCount) external onlyPDPVerifier {
         receiveProofSetEvent(proofSetId, OperationType.PROVE_POSSESSION, abi.encode(challengedLeafCount, seed, challengeCount));
-        emit Debug("Here we go", 0);
         if (provenThisPeriod[proofSetId]) { 
-            // return immediately, we've already witnessed a proof for this proof set this period
-            return; 
+            revert("Only one proof of possession allowed per proving period. Open a new proving period."); 
         }
-
         if (challengeCount < getChallengesPerProof()) {
             revert("Invalid challenge count < 5");
         }
-        emit Debug("deadline", provingDeadlines[proofSetId]);
-        emit Debug("block number", block.number);
-        // check for late proof 
+        // check for proof outside of proving period
         if (provingDeadlines[proofSetId] < block.number) {
-            uint256 periodsLate = 1 + ((block.number - provingDeadlines[proofSetId]) / getMaxProvingPeriod());
-            emit Debug("we're late", periodsLate); 
-            emit FaultRecord(FaultType.LATE, periodsLate);
+            revert("Current proving period passed. Open a new proving period.");
+        } 
+        if (provingDeadlines[proofSetId] - getMaxProvingPeriod() >= block.number) {
+            revert("Too early. Wait for proving period to open");
         }
         provenThisPeriod[proofSetId] = true;
     }
 
-    // nextProvingPeriod checks for unsubmitted proof and emits a fault record if so
+    // nextProvingPeriod checks for unsubmitted proof and emits a fault if so
     function nextProvingPeriod(uint256 proofSetId, uint256 leafCount) external onlyPDPVerifier {
         receiveProofSetEvent(proofSetId, OperationType.NEXT_PROVING_PERIOD, abi.encode(leafCount));
-        // check for unsubmitted proof 
-        if (!provenThisPeriod[proofSetId]) {
-            uint256 periodsSkipped = 1;
-            if (provingDeadlines[proofSetId] < block.number) {
-                periodsSkipped = 1 + ((block.number - provingDeadlines[proofSetId]) / getMaxProvingPeriod());
+        // Noop when proving period not yet open
+        // Can only get here if calling nextProvingPeriod multiple times within the same proving period
+        uint256 prevDeadline = provingDeadlines[proofSetId] - getMaxProvingPeriod();
+        if (block.number <= prevDeadline) {
+            return;
```

Comment:
I think this needs to revert. As is, a provider could grind challenges by calling `verifier.nextProvingPeriod` once each epoch after a successful proof until the deadline is reached.

In general, I think any call to this method must either (1) advance the proving period (possibly with fault), or (2) revert.

Reply:
This makes me think that, in the PDP Verifier contract, the method name `nextProvingPeriod`  is not quite right. That contract doesn't know anything about proving periods. It could instead be `newChallenge` or something. However since that's the method actually invoked by SPs, I can see the motivation for a name that represents their expected workflow.

Reply:
Thanks for the catch on the revert.

Yeah I'm still not happy with the name but there is some sort of proving period thing that the pdp verifier knows about.  Namely when batched adds and removes get added into the proofset.  I think this name is probably ok for now even though pdp service has a modified idea of proving period